VL.Data/Services/BookService.cs

```diff
@@ -0,0 +1,63 @@
+﻿using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using VL.Shared.Data;
+using VL.Shared.Interfaces;
+using VL.Shared.Model;
+
+namespace VL.Shared.Services
+{
+    public class BookService : IBookService
+    {
+        private readonly ApplicationDbContext _applicationDbContext;
+        private readonly IMapper _mapper;
+
+        public BookService(ApplicationDbContext applicationDbContext, IMapper mapper)
+        {
+            _applicationDbContext = applicationDbContext;
+            _mapper = mapper;
+        }
+
+        public async Task<IEnumerable<BookDto>> GetBooksAsync()
+        {
+            var bookList = await _applicationDbContext.Book
+                .ToListAsync();
+            var mappedBookList = bookList.Select(bookList => _mapper.Map<BookDto>(bookList));
+            return mappedBookList;
+        }
+
+        public async Task<BookDto?> GetBookAsync(int id)
+        {
+            var book = await _applicationDbContext.Book.FindAsync(id);
+            if (book == null) return null;
+            var mappedBook = _mapper.Map<BookDto>(book);
+            return mappedBook;
+        }
+
+        public async Task<Book?> UpdateBookAsync(BookDto bookDto)
+        {
+            var mappedBook = _mapper.Map<Book>(bookDto);
+            _applicationDbContext.Book.Update(mappedBook);
+            await _applicationDbContext.SaveChangesAsync();
+            return mappedBook;
+        }
+
+        public async Task<Book> CreateBookAsync(BookDto bookDto)
+        {
+            var mappedBook = _mapper.Map<Book>(bookDto);
```

Comment:
Don't invent variables, just use the type (especially when using var)

`mappedBook` -> `bookDto`

Reply:
same as above

Reply:
If it is a BookDto and you're converting it to a Book

variable naming would be `bookDto` -> `book`

If it is a Book and you're converting it to a BookDto

variable naming would be `book` -> `bookDto`

If it is even required in your logic, most of the times you can just return the mapped type and never assign it to a variable

this is the way